File: app/src/main/java/com/test/xyz/demo/domain/repository/api/WeatherRepository.java
package com.test.xyz.demo.domain.repository.api;

import com.test.xyz.demo.domain.model.weather.Result;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface WeatherRepository {
    String HTTPS_API_WEATHER_URL = "https://api.openweathermap.org/";

    @GET("data/2.5/weather")
    Observable<Result> getWeatherInfo(@Query("q") String query, @Query("APPID") String appID, @Query("units") String units);
}

File: app/src/main/kotlin/com/test/xyz/demo/domain/repository/api/WeatherRepository.kt
package com.test.xyz.demo.domain.repository.api

import com.test.xyz.demo.domain.model.weather.Result

import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query

interface WeatherRepository {

    @GET("data/2.5/weather")
    fun getWeatherInfo(@Query("q") query: String, @Query("APPID") appID: String, @Query("units") units: String): Observable<Result>

    companion object {
        val HTTPS_API_WEATHER_URL = "https://api.openweathermap.org/"
    }
}
